app/models/text_field_type.rb

```diff
@@ -25,14 +29,23 @@ def mapping_field_name
   end
 
   def text_present
-    errors.add(:text, "must be present") if @text.empty?
+    errors.add(:text, 'must be present') if @text.empty?
   end
 
   def text_length
     validator = LengthValidator.new(validations[:length].merge(attributes: [:text]))
     validator.validate_each(self, :text, text)
   end
 
+  def text_unique
+    errors.add(:base, 'Slug Must be unique') unless FieldItem.jsonb_contains(:data, text: text).empty?
```

Comment:
Again, this must be scoped to a Field. If you have any questions about this, we can discuss. See my previous comment: https://github.com/cbdr/cortex/pull/411#discussion_r93269406